File: complicaus_2/osumania.py
# Source: Codeforces
# URL: https://codeforces.com/problemset/problem/2009/B
# Solution by Kenny Jesús Flores Huamán

t = int(input())
for _ in range(t):
    n = int(input())
    res = []
    for _ in range(n):
        row = input()
        target = row.index('#') + 1

        res = [target] + res
    
    sol = ' '.join(map(str, res))
    print(sol)

File: complicaus_2/osumania.hs
-- Source: Codeforces
-- URL: https://codeforces.com/problemset/problem/2009/B
-- Solution by Alejandro Domínguez Muñoz

{-# LANGUAGE LambdaCase #-}

import Data.List
import Data.Maybe

groupJust = fmap catMaybes . filter (/= [Nothing]) . groupBy (\_ x -> isJust x)

note = \case
    "#..." -> Just 1
    ".#.." -> Just 2
    "..#." -> Just 3
    "...#" -> Just 4
    _      -> Nothing

run = fmap reverse . groupJust . fmap note . lines

main = interact $ unlines . fmap (unwords . fmap show) . run
